config/types/etcd.go

```diff
@@ -142,6 +145,39 @@ func etcdContents(etcd Etcd, platform string) (string, error) {
 	return assembleUnit("/usr/lib/coreos/etcd-wrapper $ETCD_OPTS", args, vars, platform)
 }
 
+var validPrefixes = []string{"http://", "https://", "unix://", "unixs://"}
+
+func checkForPrefix(s string) bool {
+	hasPrefix := false
+	for _, p := range validPrefixes {
+		hasPrefix = hasPrefix || strings.HasPrefix(s, p)
+	}
+	return hasPrefix
+}
+
+func (e Etcd3_0) Validate() report.Report {
+	r := report.Report{}
+	addPrefixWarning := func(name string) {
+		r.Add(report.Entry{
+			Message: fmt.Sprintf("%s doesn't start with one of: %v", name, validPrefixes),
+			Kind:    report.EntryWarning,
+		})
+	}
+	if !checkForPrefix(e.ListenPeerUrls) {
```

Comment:
```go
for key, attr in map[string]string {
    "listen_peer_urls":            e.ListenPeerUrls,
    "listen_client_urls":          e.ListenClientUrls,
    "initial_advertise_peer_urls": e.InitialAdvertisePeerUrls,
    "advertise_client_urls":       e.AdvertiseClientUrls,
} {
    if !checkForPrefix(attr) {
        r.Add(report.Entry{
            Message: fmt.Sprintf("value for %s doesn't start with one of: %v", prefix, validPrefixes),
            Kind:    report.EntryWarning,
        })
    }
}
```

You probably want to put that in its own function to avoid duplication between the different Validate methods.